tutorials/kubernetes-auth-openid-rbac.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Kubernetes 1.6.1 Authentication via Google OpenID
+description: A sample process how to use Google Accounts with Kubernetes cluster with Role Based Access-Control (RBAC) authorization mode.
+author: miry
+tags: Kubernetes
+date_published: 2017-05-10
+---
+
+In this tutorial we will setup authentication and authorization using [RBAC] and [OpenID Connect]. [RBAC] introduced in the Kubernetes 1.6 [article](http://blog.kubernetes.io/2017/04/rbac-support-in-kubernetes.html). The article based on Micah Hausle’s [Reduce administrative toil with Kubernetes 1.3](https://www.skuid.com/blog/reduce-administrative-toil-with-kubernetes-1-3/)
+
+### Objectives
+- Creating a Google API Console project and client ID
+- Setup a Kubernetes cluster with kubeadm
+- Generate a local user credentials
+- Grant permissions
+
+We need a Kubernetes cluster to work setup via [kubeadm].
+
+## Step 1: Creating a Google API Console project and client ID
+- Go to https://console.developers.google.com/projectselector/apis/library
+- From the project drop-down, select an existing project, or create a new one by selecting **Create a new project**
+- In the sidebar under **API Manager**, select **Credentials**, then select the **OAuth consent screen** tab.
+- Choose an **Email Address**, specify a **Product Name**, and submit **Save**.
+- In the **Credentials** tab, select the **New credentials** drop-down list, and choose **OAuth client ID**.
+- Under **Application type**, select **Other**.
+- From the resulting **OAuth client dialog box**, copy the **Client ID**. The **Client ID** lets your app access enabled Google APIs.
+- Download the client secret JSON file of the credentials.
+
+## Step 2: Setup a Kubernetes cluster
+After we initialized the master instance, need to update the `kube api server` arguments in the `/etc/kubernetes/manifests/kube-apiserver.yaml`. Each argument should be on a separate line. More details about OIDC attributes you can find in the references [Authenticating](https://kubernetes.io/docs/admin/authentication/#option-1---oidc-authenticator).
+
+```
+$ sed -i "/- kube-apiserver/a\    - --oidc-issuer-url=https://accounts.google.com\n    - --oidc-username-claim=email\n    - --oidc-client-id=<Your Google Client ID>" /etc/kubernetes/manifests/kube-apiserver.yaml
```

Comment:
hm, I think I need provide a sample output, how it should be after the command. Because the file format `yaml` use indents and new lines for structure elements i added new lines.